server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java

```diff
@@ -518,6 +572,11 @@ void updateErrorState(ErrorState errorState) {
             return;
         }
 
+        if (errorState.projectId().isPresent() && clusterService.state().metadata().hasProject(errorState.projectId().get()) == false) {
```

Comment:
This was a bug. If parsing fails for a project that doesn't exist yet, the error state reporting won't work since it's persisted in the project metadata that doesn't exist yet. This case needs to be handled in the caller (multi-project file service).